trino/client.py

```diff
@@ -776,6 +701,11 @@ def fetch(self) -> List[List[Any]]:
         self._response_headers = response.headers
         if status.next_uri is None:
             self._finished = True
+
+        if self._experimental_python_types:
+            factory = RowMapperFactory(status.columns)
```

Comment:
`RowMapperFactory().create(...)` should only be called once when the columns are known. Maybe move this logic to the `_update_state` method. Add the created `RowMapper` instance on an instance field of `TrinoQuery`.

Reply:
I modified the code as best as I could, because in `execute()` function expects some row (maybe) yet the columns may be not known yet.

Reply:
Yeah, the columns are only known when Trino is starting to send data, which can be in the first `execute` or in any of the subsequent `fetch` calls. We should delay the creation of the `RowMapper` until the columns are present and reuse that cached `RowMapper` on the fetched data to get the performance benefit.